src/Interpreters/Streaming/SubstituteStreamingFunction.cpp

```diff
@@ -77,21 +78,25 @@ std::optional<String> StreamingFunctionData::supportChangelog(const String & fun
     /// Support combinator suffix, for example:
     /// `count`                 => `__count_retract`
     /// `count_if`              => `__count_retract_if`
-    /// `count_distinct`        => `__count_retract_distinct`
-    /// `count_distinct_if`     => `__count_retract_distinct_if`
+    /// `count_distinct`        => `__count_retract_distinct_retract`
+    /// `count_distinct_if`     => `__count_retract_distinct_retract_if`
     String combinator_suffix;
     auto nested_func_name = function_name;
     while (iter == changelog_func_map.end())
     {
         if (auto combinator = AggregateFunctionCombinatorFactory::instance().tryFindSuffix(nested_func_name))
         {
-            const std::string combinator_name = combinator->getName();
+            std::string combinator_name = combinator->getName();
             /// TODO: support more combinators
-            if (combinator_name != "_if")
+            if (combinator_name != "_if" && combinator_name != "_distinct" && combinator_name != "_distinct_retract")
```

Comment:
not need care `_distinct_retract` ， can't happen here

Reply:
what if user input `_distinct_retract` ? do not consider this condition?

Reply:
we should block them

Reply:
yes, let's keep your changes